File: c#/src/base/Exchange.String.cs
namespace ccxt;

public partial class Exchange
{

    public string uuid2()
    {
        return Guid.NewGuid().ToString();
    }

    public string uuid()
    {
        return Guid.NewGuid().ToString();
    }

    public string uuid16()
    {
        return Guid.NewGuid().ToString().Substring(0, 16);
    }

    public string uuid22()
    {
        return Guid.NewGuid().ToString();
    }

    public object strip(object str)
    {
        return str; //stub
    }

    public string capitalize(object str2)
    {
        var str = (string)str2;
        return char.ToLower(str[0]) + str.Substring(1);
    }


}

File: go/ccxt/exchange_string.go
package ccxt

import (
	"strings"
	"unicode"

	u "github.com/google/uuid"
)

// BaseUID generates a new UUID and returns it as a string without dashes.
func (this *Exchange) BaseUID() string {
	return strings.ReplaceAll(u.New().String(), "-", "")
}

// uuid2 generates a new UUID and returns it as a string.
func (this *Exchange) Uuid2() string {
	return u.New().String()
}

// func (this *Exchange) uuid22() string {
// 	return ""
// }

// uuid is a wrapper for uuid2.
func (this *Exchange) Uuid() string {
	return this.Uuid2()
}

func (this *Exchange) Uuidv1() string {
	return u.New().String()
}

// uuid16 returns the first 16 characters of a UUID without dashes.
func (this *Exchange) Uuid16() string {
	return this.BaseUID()[:16]
}

// uuid22 returns the first 22 characters of a UUID without dashes.
func (this *Exchange) Uuid22() string {
	return this.BaseUID()[:22]
}

// strip trims whitespace from both ends of a string.
func (this *Exchange) Strip(str interface{}) interface{} {
	return strings.TrimSpace(str.(string))
}

// capitalize capitalizes the first letter of a string.
func (this *Exchange) Capitalize(str2 interface{}) string {
	str := str2.(string)
	if len(str) == 0 {
		return str
	}
	runes := []rune(str)
	runes[0] = unicode.ToUpper(runes[0])
	return string(runes)
}
